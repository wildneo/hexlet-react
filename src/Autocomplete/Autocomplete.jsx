import axios from 'axios';
import React from 'react';

export default class Autocomplete extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      text: '',
      countries: null,
    };

    this.handleChange = this.handleChange.bind(this);
  }

  async handleChange({ target: { value } }) {
    if (value === '') {
      this.setState({ countries: null, text: '' });

      return;
    }
    const response = await axios.get('/countries', { params: { term: value } });
    this.setState({ countries: response.data, text: value });
  }

  render() {
    const { text, countries } = this.state;

    return (
      <div>
        <form>
          <div className="form-group">
            <input className="form-control" type="text" placeholder="Enter Country" value={text} onChange={this.handleChange} />
          </div>
        </form>
        {countries && <ul>{countries.map((item) => <li key={item}>{item}</li>)}</ul>}
      </div>
    );
  }
}
