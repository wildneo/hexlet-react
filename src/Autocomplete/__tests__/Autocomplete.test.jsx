import nock from 'nock';
import React from 'react';
import axios from 'axios';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import httpAdapter from 'axios/lib/adapters/http';
import timeout from 'timeout-then';
import Autocomplete from '../Autocomplete';

const host = 'http://localhost';

beforeAll(() => {
  Enzyme.configure({ adapter: new Adapter() });

  axios.defaults.adapter = httpAdapter;
  nock.disableNetConnect();
});

test('Autocomplete', async () => {
  const wrapper = mount(<Autocomplete />);
  expect(wrapper.render()).toMatchSnapshot();

  nock(host)
    .get('/countries')
    .query({ term: 'a' })
    .reply(200, ['albania', 'aljir']);

  const input = wrapper.find('input');
  input.simulate('change', { target: { value: 'a' } });
  await timeout(200);
  expect(wrapper.render()).toMatchSnapshot();
  nock(host)
    .get('/countries')
    .query({ term: 'alb' })
    .reply(200, ['albania']);
  input.simulate('change', { target: { value: 'alb' } });
  await timeout(100);
  expect(wrapper.render()).toMatchSnapshot();

  input.simulate('change', { target: { value: '' } });
  expect(wrapper.render()).toMatchSnapshot();
});
