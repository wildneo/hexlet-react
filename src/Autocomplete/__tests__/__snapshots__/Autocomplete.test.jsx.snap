// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete 1`] = `
<div>
  <form>
    <div
      class="form-group"
    >
      <input
        class="form-control"
        placeholder="Enter Country"
        type="text"
        value=""
      />
    </div>
  </form>
</div>
`;

exports[`Autocomplete 2`] = `
<div>
  <form>
    <div
      class="form-group"
    >
      <input
        class="form-control"
        placeholder="Enter Country"
        type="text"
        value="a"
      />
    </div>
  </form>
  <ul>
    <li>
      albania
    </li>
    <li>
      aljir
    </li>
  </ul>
</div>
`;

exports[`Autocomplete 3`] = `
<div>
  <form>
    <div
      class="form-group"
    >
      <input
        class="form-control"
        placeholder="Enter Country"
        type="text"
        value="alb"
      />
    </div>
  </form>
  <ul>
    <li>
      albania
    </li>
  </ul>
</div>
`;

exports[`Autocomplete 4`] = `
<div>
  <form>
    <div
      class="form-group"
    >
      <input
        class="form-control"
        placeholder="Enter Country"
        type="text"
        value=""
      />
    </div>
  </form>
</div>
`;
